packages/zwave-js/src/lib/commandclass/SceneActuatorConfigurationCC.ts

```diff
@@ -0,0 +1,395 @@
+	sceneId: number,
+	level: number,
+	dimmingDuration: Duration,
+) {
+	const levelValueId = getLevelValueID(this.endpointIndex, sceneId);
+	const dimmingDurationValueId = getDimmingDurationValueID(
+		this.endpointIndex,
+		sceneId,
+	);
+	const valueDB = this.getValueDB();
+
+	if (!valueDB.hasMetadata(levelValueId)) {
+		valueDB.setMetadata(levelValueId, {
+			...ValueMetadata.UInt8,
+			label: `Level (${sceneId})`,
+		});
+	}
+	if (!valueDB.hasMetadata(dimmingDurationValueId)) {
+		valueDB.setMetadata(dimmingDurationValueId, {
+			...ValueMetadata.Duration,
+			label: `Dimming duration (${sceneId})`,
+		});
+	}
+
+	valueDB.setValue(levelValueId, level);
+	valueDB.setValue(dimmingDurationValueId, dimmingDuration);
+
+	return true;
+}
+
+@API(CommandClasses["Scene Actuator Configuration"])
+export class SceneActuatorConfigurationCCAPI extends CCAPI {
+	public supportsCommand(
+		cmd: SceneActuatorConfigurationCommand,
+	): Maybe<boolean> {
+		switch (cmd) {
+			case SceneActuatorConfigurationCommand.Get:
+				return this.isSinglecast();
+			case SceneActuatorConfigurationCommand.Set:
+			case SceneActuatorConfigurationCommand.Report:
+				return true;
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property, propertyKey },
+		value,
+	): Promise<void> => {
+		if (propertyKey == undefined) {
+			throwMissingPropertyKey(this.ccId, property);
+		} else if (typeof propertyKey !== "number") {
+			throwUnsupportedPropertyKey(this.ccId, property, propertyKey);
+		}
+		if (property === "level") {
+			if (typeof value !== "number") {
+				throwWrongValueType(
+					this.ccId,
+					property,
+					"number",
+					typeof value,
+				);
+			}
+
+			// We need to set the dimming duration along with the level
+			const node = this.endpoint.getNodeUnsafe()!;
+			// If duration is missing, we set a default of instant
+			const dimmingDuration =
+				node.getValue<Duration>(
+					getDimmingDurationValueID(this.endpoint.index, propertyKey),
+				) ?? new Duration(0, "seconds");
+			await this.set(propertyKey, dimmingDuration);
+		} else {
+			// setting dimmingDuration value alone not supported,
+			// to be added when setting Duration values is supported
+			throwUnsupportedProperty(this.ccId, property);
+		}
+
+		// Verify the current value after a delay
+		this.schedulePoll({ property, propertyKey });
+	};
+
+	protected [POLL_VALUE]: PollValueImplementation = async ({
+		property,
+		propertyKey,
+	}): Promise<unknown> => {
+		switch (property) {
+			case "level":
+			case "dimmingDuration": {
+				if (propertyKey == undefined) {
+					throwMissingPropertyKey(this.ccId, property);
+				} else if (typeof propertyKey !== "number") {
+					throwUnsupportedPropertyKey(
+						this.ccId,
+						property,
+						propertyKey,
+					);
+				}
+				return (await this.get(propertyKey))?.[property];
+			}
+			default:
+				throwUnsupportedProperty(this.ccId, property);
+		}
+	};
+
+	public async set(
```

Comment:
Maybe we should call this `enableScene`

Reply:
I think of "enable" as being a toggle.  Something is either enabled or disabled.  Enabling something that is already enabled shouldn't change anything

In this case, the user is also adjusting duration and (optionally) level settings.

Maybe `setupScene` or `setScene` ?

Reply:
The more I think about this, the more I prefer just leaving it as `set`.  The `set` function is configuring the endpoint for how the actuator should behave when a scene is activating.  It's basically setting the scene's actuator configuration.

Reply:
Yeah, see next comment. I got confused